### ISSUE TITLE
jTracert doesn't work in OSGi environment

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ankit4u3/jtracert#14]
```
What steps will reproduce the problem?
1. Execute an OSGI application with jTracert
2. Instrumented bundles will fail to start

What is the expected output? What do you see instead?

Application under test should work fine. Instead we get a
ClassNotFoundException as below:

java.lang.ClassNotFoundException:
com.google.code.jtracert.traceBuilder.MethodCallTraceBuilderFactory
    at
org.eclipse.osgi.framework.internal.core.BundleLoader.findClass(BundleLoader.jav
a:402)
    at
org.eclipse.osgi.framework.internal.core.BundleLoader.findClass(BundleLoader.jav
a:347)
    at
org.eclipse.osgi.internal.baseadaptor.DefaultClassLoader.loadClass(DefaultClassL
oader.java:83)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:251)
    at java.lang.ClassLoader.loadClassInternal(ClassLoader.java:319)
    at com.vero.ipnetro.server.Activator.<clinit>(Activator.java)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at
sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorI
mpl.java:39)
    at
sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorA
ccessorImpl.java:27)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:513)
    at java.lang.Class.newInstance0(Class.java:355)
    at java.lang.Class.newInstance(Class.java:308)
    at
org.eclipse.osgi.framework.internal.core.AbstractBundle.loadBundleActivator(Abst
ractBundle.java:136)
    at
org.eclipse.osgi.framework.internal.core.BundleContextImpl.start(BundleContextIm
pl.java:962)
    at
org.eclipse.osgi.framework.internal.core.BundleHost.startWorker(BundleHost.java:
317)
    at
org.eclipse.osgi.framework.internal.core.AbstractBundle.start(AbstractBundle.jav
a:256)
    at
org.eclipse.osgi.framework.internal.core.FrameworkCommandProvider._start(Framewo
rkCommandProvider.java:239)
    at sun.reflect.GeneratedMethodAccessor1.invoke(Unknown Source)
    at
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.jav
a:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at
org.eclipse.osgi.framework.internal.core.FrameworkCommandInterpreter.execute(Fra
meworkCommandInterpreter.java:145)
    at
org.eclipse.osgi.framework.internal.core.FrameworkConsole.docommand(FrameworkCon
sole.java:293)
    at
org.eclipse.osgi.framework.internal.core.FrameworkConsole.console(FrameworkConso
le.java:278)
    at
org.eclipse.osgi.framework.internal.core.FrameworkConsole.run(FrameworkConsole.j
ava:213)
    at java.lang.Thread.run(Thread.java:619)

```

Original issue reported on code.google.com by `Dmitry.Bedrin` on 5 Mar 2009 at 8:47

**Comment by GoogleCodeExporter** [comment]
```
One of jTracert users has reported that he's experiencing problems with 
analyzing
Eclipse plugins with jTracert.

Since Eclipse is also based on OSGi these defects may be closely related.
```

Original comment by `Dmitry.Bedrin` on 11 Mar 2009 at 12:19

**Comment by GoogleCodeExporter** [comment]
```
On JRE 1.6+ jTracert is now appends it classes to the bootstrap class path - it
solves the issue.

Fixed in SVN revision r215

However, we still need to support JRE 1.5, so there're still things to 
implement.
```

Original comment by `Dmitry.Bedrin` on 13 Mar 2009 at 2:18
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in jTracert version 0.1.1
```

Original comment by `Dmitry.Bedrin` on 17 Mar 2009 at 10:12

**Comment by GoogleCodeExporter** [comment]
```
JRE 1.5 support is added in SVN revision r276
To be released in version 0.1.2
```

Original comment by `Dmitry.Bedrin` on 29 Apr 2009 at 12:32
